Document k6/experimental/redis module (#768)
grafana/k6-docs: src/data/markdown/docs/02 javascript api/13 k6-redis/10 Client.md

---
title: 'Client'
excerpt: 'Client is a Redis client to interact with a Redis server or cluster.'
---

> `Client` is a Redis client to interact with a Redis server or cluster. It exposes a promise-based API, allowing users to interact with Redis in an asynchronous manner. The `Client` API intends to be quite thorough and extensive, but is still limited to a set of functionalities the k6 team judged useful in the context of k6. The intent of the `Client` isn't to provide a full-fledged Redis client, exposing the whole Redis API, but rather to empower our users to use Redis in use-cases we believe are relevant to k6.

Reviewer: ```suggestion
`Client` is a Redis client to interact with a Redis server or cluster. It exposes a promise-based API, which users can interact with in an asynchronous manner.

Though the `Client` API intends to be thorough and extensive, it does not expose the whole Redis API.
Instead, the API aims to empower users to use Redis for use cases most appropriate to k6. 
```

> Note that the configuration of the `Client` is done through the [`Options`](/javascript-api/k6-redis/options) object.

Reviewer: ```suggestion
Note that you configure the `Client` through the [`Options`](/javascript-api/k6-redis/options) object.
```
